projects/core/src/product/services/category-page-title.resolver.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { Observable, of } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { RoutingService } from '../../routing/facade/routing.service';
+import { CmsService } from '../../cms/facade/cms.service';
+import { Page } from '../../cms/model/page.model';
+
+import { PageType } from '../../occ/occ-models/occ.models';
+import { PageTitleResolver } from '../../cms/page/page-title.resolver';
+import { ProductSearchService } from '../facade/product-search.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CategoryPageTitleResolver extends PageTitleResolver {
+  constructor(
+    protected routingService: RoutingService,
+    protected productSearchService: ProductSearchService,
+    protected cms: CmsService
+  ) {
+    super();
+    this.pageType = PageType.CATEGORY_PAGE;
+  }
+
+  resolve(): Observable<string> {
+    return this.cms.getCurrentPage().pipe(
```

Comment:
In other places we filter the `getCurrentPage` method with `Boolean`, so probably we should also add it here.

I suggest to perform `filter(Boolean)` directly in the implementation of `getCurrentPage` - in scope of another ticket.

Reply:
Agreed. We had an issue with smartedit integration and therefor haven't introduced it lower level.